Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -194,31 +198,59 @@ class MessageFieldGenerator: FieldGeneratorBase, FieldGenerator {
             traitsArg = ""
         }
 
-        let varName = hasFieldPresence ? "v" : storedProperty
-
         var usesLocals = false
-        let conditional: String
+        let conditionals: [(condition: String, varName: String)]
         if isRepeated {  // Also covers maps
-            conditional = "!\(varName).isEmpty"
+            conditionals = [
+                (condition: "visitDefaultValues || !\(storedProperty).isEmpty", varName: storedProperty)
```

Comment:
Realizing this as a total nit pick: should we flip this conditional?  The majority case is visitDefaultValues will be false, so should we check it second?

Reply:
Good callout, updated